EEP_Gleisplan.html

```diff
@@ -751,39 +764,85 @@
-		const PosY			= -Gleis.getElementsByTagName('Dreibein')[0].getElementsByTagName('Vektor')[0].getAttribute('y') / 100;
-
-		const width 		= global_panZoom.getSizes().width;
-		const height		= global_panZoom.getSizes().height;
-		const realZoom		= global_panZoom.getSizes().realZoom;
-		
-		const panX 			= width/2	- PosX * realZoom;
-		const panY 			= height/2	- PosY * realZoom;
+		if (!Gleis) { 
+			msg(`${GleissystemText[GleissystemID]} ${GleisID} existiert nicht`);
+			return false; 
+		}
 
-		global_panZoom.pan({x: panX, y: panY,})	
+		centerGleis(Gleis);
+		msg(`Positionierung auf ${GleissystemText[GleissystemID]} ${GleisID}`);
+//console.log(`Gleis ${GleissystemID}-${GleisID} Pos: (${PosX} ${PosY}) Pan: (${panX} ${panY})`)
+		return true;
 		
 	} else if (sel == 'W') {		// Weiche (Nummern übergreifend über alle Gleissysteme, jedoch nicht eindeutig)
 		// Gleis.getAttribute('Key_Id')
-		const Key_Id 		= val;
+		const WeicheID 		= val;
 		//...
 		
 	} else if (sel == 'S') {		// Signal (Nummern übergreifend über alle Gleissysteme)
 		// Gleis.getAttribute('Meldung').getAttribute('Key_Id')
-		const Key_Id 		= val;
-		//...
+		const SignalID 		= val;
+		
+		// Suche das Signal auf allen Gleisen
+		const Gleissysteme 	= global_sutrackp.getElementsByTagName('Gleissystem')
+		for (const Gleissystem of Gleissysteme) {
+			
+			const Gleise = Gleissystem.getElementsByTagName('Gleis');
+			for (const Gleis of Gleise) {
+
+				const Meldungen = Gleis.getElementsByTagName('Meldung');
+				for (const Meldung of Meldungen) {
+
+					const Key_Id	= Meldung.getAttribute('Key_Id'); 	// Id eines Signals
+					if (Key_Id == SignalID) {
+						// Signal gefunden
+						const GleissystemID = Gleissystem.getAttribute('GleissystemID');
+						const GleisID = Gleis.getAttribute('GleisID');
+						centerGleis(Gleis);
+						msg(`Das Signal ${SignalID} befindet sich auf ${GleissystemText[GleissystemID]} ${GleisID}`);
+//console.log(`Gleis ${GleissystemID}-${GleisID} Pos: (${PosX} ${PosY}) Pan: (${panX} ${panY})`)
+						return true;	
+					}
+				
+				} // ´Meldungen
+			} // Gleise
+		} // Gleissysteme
+		msg(`Signal ${SignalID} existiert nicht`);
+		return false; // Signal nicht gefunden
 
 	} else if (sel == 'K') {		// Kontakt
 		//...
 	}
 	return false;
 }
 
+function centerGleis(Gleis) {
+	const PosX		= Gleis.getElementsByTagName('Dreibein')[0].getElementsByTagName('Vektor')[0].getAttribute('x') / 100;
+	const PosY		= -Gleis.getElementsByTagName('Dreibein')[0].getElementsByTagName('Vektor')[0].getAttribute('y') / 100;
+
+	const width 	= global_panZoom.getSizes().width;
+	const height	= global_panZoom.getSizes().height;
+	const realZoom	= global_panZoom.getSizes().realZoom;
+	
+	const panX 		= width/2	- PosX * realZoom;
+	const panY 		= height/2	- PosY * realZoom;
+
+	global_panZoom.pan({x: panX, y: panY,});
+}
+
+function msg(text) {
+	if (text) {
+		document.getElementById('message').innerHTML = text;
```

Comment:
If the `text` does not contain HTML use `textContent` for performance (no layout invalidation) and security.

Reply:
done